kratos/solving_strategies/builder_and_solvers/explicit_builder.h

```diff
@@ -593,8 +593,7 @@ class ExplicitBuilder
         const int n_constraints = static_cast<int>(r_constraints_array.size());
 
         // Global dof set
-        DofSetType dof_global_set;
-        dof_global_set.reserve(n_elems*20);
+        mDofSet = DofsArrayType();
```

Comment:
you need to first push it back onto an unordered one, and then add it at once to the ordered array